aggregator_core/src/datastore.rs

```diff
@@ -90,6 +96,8 @@ macro_rules! supported_schema_versions {
 
 // List of schema versions that this version of Janus can safely run on. If any other schema
 // version is seen, [`Datastore::new`] fails.
+//
+// Note that the latest supported version must be first in the list.
 supported_schema_versions!(9);
```

Comment:
Experimenting to see if this can be asserted statically. The logic would be `assert!(max(SUPPORTED_SCHEMA_VERSIONS) == SUPPORTED_SCHEMA_VERSIONS[0])`, but I don't see a const way of getting the max of a slice.

If it's too complicated I'll just add a standard test, just would be nice to have errors like that caught at compile-time.

Reply:
Ah wait, https://doc.rust-lang.org/std/cmp/fn.max.html `std::cmp::max` takes a `T: Ord`, which arrays do implement https://doc.rust-lang.org/std/primitive.array.html.

Reply:
Eh, but it's unstable, and unstable in a somewhat scary way https://github.com/rust-lang/rust/issues/92391#issuecomment-1492469098

Reply:
Sorry, I hadn't thought my suggestion all the way through. I think it'll be another few years before enough trait implementations are available in const contexts to do a static assertion -- a plain test should be fine here.